Add diagnostics sets to tutorial

forsyth2 opened the pull request:
Add diurnal cycle and streamflow diagnostics sets to tutorial. Resolves #378. Expanding on #369.

chengzhuzhang commented:
Hi Ryan, we had a duplicated PR:https://github.com/E3SM-Project/e3sm_diags/pull/369
Are you thinking something similar for updating tutorial?

forsyth2 commented:
@chengzhuzhang Oh I forgot about that PR. That looks very similar to the tutorial `examples/tutorials/tutorial_2020_all_sets.py `.  Do you intend to replace that file with `examples/tutorials/run_v230_all_sets.py` in #369? If that's the case, I can ignore that file.

In any case, we need to add the new diagnostics sets to `tests/test_complete_run.py`, which I'm also doing for this PR.

chengzhuzhang commented:
It seems like your PR has more changes, so please go ahead. I would suggest to have a new script with 'latest' in the file name rather than update the tutorial2020 file though.

forsyth2 commented:
@chengzhuzhang This is ready for review. I'm not sure if you want to close #369 then. The tutorial and accompanying test now include diurnal cycle and streamflow. I also did some refactoring to reduce code duplication (tutorial parameters only specified in one place now; image comparison code is only in one place now). 

The results of running the tutorial on Cori are at https://portal.nersc.gov/project/e3sm/forsyth/v2_3_0_all_sets/viewer/. 

The list of expected images is at https://web.lcrc.anl.gov/public/e3sm/e3sm_diags_test_data/unit_test_complete_run/expected_images_v2_3_0_all_sets_2021_02_09.txt. Running the test on Anvil took ~647 seconds.

@tomvothecoder I ran the unit tests too. All tests pass, taking ~205 seconds. As you mention in #385, merging this pull request will fix the file name error causing the tests to fail (`test_complete_run.py` is now `complete_run_v2_3_0.py`).

chengzhuzhang commented on tests/complete_run_v2_3_0.py:
```diff
@@ -0,0 +1,38 @@
+sys.path.insert(0, '../examples')
+from tutorials.tutorial_v2_3_0_all_sets import run_anvil
+
+# Due to the large amount of data required to run, this test will be run manually on Anvil
```
Since this script will be run manually, it will be handy to add instruction on running on compute nodes and activating e3sm-unified env.

chengzhuzhang reviewed (commented):
The PR looks good, Ryan. The `tutorial_v2_3_0_all_sets.py` will serve as an excellent script for our internal testing or examples for users with E3SM machine accounts. I would avoid a run script getting too long for general use though. So a couple of suggested changes:
* rename `tutorial_v2_3_0_all_sets.py` to `tutorial_v2_3_0_all_sets_E3SM_machines.py`
* add run_compy
* since it already has instruction on activating `e3sm-unified`, adding short instruction on running on compute nodes will be handy
* create a shorter  `tutorial_v2_3_0_all_sets.py` script ( I will take care of this)

forsyth2 commented:
@chengzhuzhang Thanks, I addressed your comments in the second commit "address comments". I ran the tutorial on Compy, LCRC, and NERSC and everything looks fine. The LCRC test also passes. Let me know if the second commit looks good, and if so, I'll squash the commits and merge.

chengzhuzhang reviewed (approved):
Looks great. Please go ahead and merge.

forsyth2 commented:
@tomvothecoder This pull request has been merged, so the file name issue has been fixed. The tests on this branch pass.
